src/sentry/integrations/slack/message_builder/notifications.py

```diff
@@ -1,69 +1,119 @@
-    settings_url = get_settings_url(notification)
+def build_notification_title(notification: BaseNotification) -> Any:
+    if isinstance(notification, AlertRuleNotification):
+        title_str = "Issue triggered"
 
-    if not notification.group:
-        # Groups are not associated with a deploy notification so in this one
-        # case, the footer is different.
-        return f"<{settings_url}|Notification Settings>"
+        if notification.rules:
+            rule_url = build_rule_url(
+                notification.rules[0], notification.group, notification.group.project
+            )
+            title_str += f" <{rule_url}|{notification.rules[0].label}>"
 
-    group_url = get_group_url(notification)
-    short_id = notification.group.qualified_short_id
-    return f"<{group_url}|{short_id}> via <{settings_url}|Notification Settings>"
+            if len(notification.rules) > 1:
+                title_str += f" (+{len(notification.rules) - 1} other)"
+
+        return title_str
+
+    return notification.get_notification_title()
 
 
 class SlackNotificationsMessageBuilder(SlackMessageBuilder):
-    def __init__(self, notification: BaseNotification, context: Mapping[str, Any]) -> None:
+    def __init__(
+        self,
+        notification: BaseNotification,
+        context: Mapping[str, Any],
+        recipient: Union[Team, User],
+    ) -> None:
         super().__init__()
         self.notification = notification
         self.context = context
+        self.recipient = recipient
 
     def build(self) -> SlackBody:
-        if isinstance(self.notification, AlertRuleNotification):
+        if isinstance(self.notification, ISSUE_UNFURL):
             return SlackIssuesMessageBuilder(
-                self.notification.group,
-                self.notification.event,
-                self.context["tags"],
-                self.notification.rules,
-                issue_alert=True,
+                group=self.notification.group,
+                event=getattr(self.notification, "event", None),
+                tags=self.context.get("tags", None),
+                rules=getattr(self.notification, "rules", None),
+                issue_details=True,
+                notification=self.notification,
+                recipient=self.recipient,
             ).build()
 
+        if isinstance(self.notification, NoteActivityNotification):
+            title_text = build_attachment_title(self.notification.group)
+            title_link = get_title_link(
+                self.notification.group, None, False, False, self.notification
+            )
+            formatted_title = f"<{title_link}|{title_text}>"
+            return self._build(
+                title=formatted_title,
+                text=self.context["text_description"],
+                footer=build_notification_footer(self.notification, self.recipient),
+            )
+
+        if isinstance(self.notification, ReleaseActivityNotification):
+            text = ""
```

Comment:
I'd move this code block to a `get_message_description()` method in `ReleaseActivityNotification`. You could make it return `self.context["text_description"]` for all other types. After that you should be able to handle every case with 
```python
return self._build(
    text=self.get_message_description(),
    footer=build_notification_footer(self.notification, self.recipient),
)
```